### DISCUSSION TITLE
(aws-dynamodb): I want to convert cdk code without deleting data in global table

### DISCUSSION THREAD

**Discussion by SGA-Kenichi-Odashima** [aws/aws-cdk#23329] [General]
### Describe the feature

I think it can be created with the Use Case API for creating a DynamoDB replica table, but is there a function to change the code that uses the API in item 1 to the code that uses the API in item 2?
Since the DynamoDB table is in operation, I would like to convert it without deleting the data.
Please let me know if there is a way to change it or if there is a function.
If there is no method or function that can be changed, I would like a function to convert like this.

### Use Case

1. aws_cdk.aws_dynamodb/CfnGlobalTable
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_dynamodb/CfnGlobalTable.html
* Sample Code (If the deployment region is us-west-2)
```
#!/usr/bin/env python3
#
# -*- coding: utf-8 -*-

from constructs import Construct
import aws_cdk as core
from aws_cdk import aws_dynamodb as dynamodb

import os

class GlobalTableStack2(core.Stack):
    def __init__(self, scope: Construct, id: str, configs: object, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        cfn_global_table = dynamodb.CfnGlobalTable(self, "dev-my-table-global",
            attribute_definitions=[dynamodb.CfnGlobalTable.AttributeDefinitionProperty(
                attribute_name="PK",
                attribute_type="S"
            ),
            dynamodb.CfnGlobalTable.AttributeDefinitionProperty(
                attribute_name="SK",
                attribute_type="S"
            )],
            key_schema=[dynamodb.CfnGlobalTable.KeySchemaProperty(
                attribute_name="PK",
                key_type="HASH"
            ),
            dynamodb.CfnGlobalTable.KeySchemaProperty(
                attribute_name="SK",
                key_type="RANGE"
            )],
            replicas=[dynamodb.CfnGlobalTable.ReplicaSpecificationProperty(
                region="us-west-2",
        
                # the properties below are optional
                contributor_insights_specification=dynamodb.CfnGlobalTable.ContributorInsightsSpecificationProperty(
                    enabled=False
                ),
            ),
            dynamodb.CfnGlobalTable.ReplicaSpecificationProperty(
                region="ap-northeast-1",
        
                # the properties below are optional
                contributor_insights_specification=dynamodb.CfnGlobalTable.ContributorInsightsSpecificationProperty(
                    enabled=False
                ),
            )],
            billing_mode="PAY_PER_REQUEST",
            stream_specification=dynamodb.CfnGlobalTable.StreamSpecificationProperty(
                stream_view_type="NEW_IMAGE"
            ),
            table_name="dev-my-table-global",
        )
        cfn_global_table.apply_removal_policy(core.RemovalPolicy.RETAIN)
```
2. aws_cdk.aws_dynamodb/Table の replication_regions
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_dynamodb/Table.html
* Sample Code (If the deployment region is us-west-2)
```
#!/usr/bin/env python3
#
# -*- coding: utf-8 -*-

from constructs import Construct
import aws_cdk as core
from aws_cdk import aws_dynamodb

import os

class GlobalTableStack(core.Stack):
    def __init__(self, scope: Construct, id: str, configs: object, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        global_table = aws_dynamodb.Table(
            self, "dev-my-table-global",
            table_name="dev-my-table-global",
            partition_key=aws_dynamodb.Attribute(
                name="PK",
                type=aws_dynamodb.AttributeType.STRING
            ),
            sort_key=aws_dynamodb.Attribute(
                name="SK",
                type=aws_dynamodb.AttributeType.STRING
            ),
            replication_regions=["ap-northeast-1"],
            point_in_time_recovery=True,
            billing_mode=aws_dynamodb.BillingMode.PAY_PER_REQUEST,
            removal_policy=core.RemovalPolicy.RETAIN
        )
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.54.0 (build 9f41881)

### Environment details (OS name and version, etc.)
Node.js Version: v16.15.1
OS: 
Language: Python

**Comment by peterwoodworth** [comment]
I don't think there is, nor could there be any function that does this. These two different apps will generate different templates with different types of resources ([`Table`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html) vs [`GlobalTable`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-globaltable.html)). CloudFormation doesn't offer a way to convert one of these types of tables to the other, so you will have to destroy the first set of tables for the other to be created.

I'm going to convert this to a discussion, as that's going to be the best place to further discuss workarounds or have further discussion. Thanks!
